library/src/main/java/org/apache/apex/malhar/lib/fs/redshift/RedshiftJdbcTransactionableOutputOperator.java

```diff
@@ -0,0 +1,283 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.apex.malhar.lib.fs.redshift;
+
+import java.sql.BatchUpdateException;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+import javax.validation.constraints.NotNull;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.apex.malhar.lib.fs.FSRecordCompactionOperator;
+
+import com.google.common.base.Preconditions;
+
+import com.datatorrent.api.Context;
+import com.datatorrent.lib.db.jdbc.AbstractJdbcTransactionableOutputOperator;
+
+/**
+ *  A generic implementation of AbstractJdbcTransactionableOutputOperator for Redshift which takes FSRecordCompactionOperator.OutputMetaData.
+ *  Load the data into the specified redshift tables from data files. The files can be located either in S3 or an Amazon EMR.
+ *  Specify the bucketName property if the file is located in S3 or specify the emrClusterId if the file is location in EMR.
+ *
+ * @displayName Redshift Output Operator
+ * @category Output
+ * @tags database, jdbc, redshift
+ */
+public class RedshiftJdbcTransactionableOutputOperator extends AbstractJdbcTransactionableOutputOperator<FSRecordCompactionOperator.OutputMetaData>
+{
+  private static final Logger logger = LoggerFactory.getLogger(RedshiftJdbcTransactionableOutputOperator.class);
+  @NotNull
+  private String tableName;
+  @NotNull
+  private String accessKey;
+  @NotNull
+  private String secretKey;
+  @NotNull
+  private String redshiftDelimiter;
```

Comment:
Is delimiter mandatory ? if yes, we must assign a default and mention that in the documentation